staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -36,6 +37,15 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
+// StartUntil starts f in a new goroutine and calls done once f has finished.
+func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
```

Comment:
Would you consider aiding in the currying by adding a function like the following?
```go
func BindChannel(f func(<-chan struct{}), c <-chan struct{}) func() {
	return func() {
		f(c)
	}
}
```
You could then write the following:
```go
RunWithinGroup(BindChannel(f, stop), wg)
```
If you really are settled on the signature, another name to consider is `StartWithinGroup`.